src/tallies/tally_scoring.cpp

```diff
@@ -0,0 +1,2347 @@
+  // rate. Otherwise, the sum of all nu-fission rates would be ~1.0.
+
+  // loop over number of particles banked
+  for (auto i = 0; i < p->n_bank_; ++i) {
+    auto i_bank = simulation::fission_bank.size() - p->n_bank_ + i;
+    const auto& bank = simulation::fission_bank[i_bank];
+
+    // get the delayed group
+    auto g = bank.delayed_group;
+
+    // determine score based on bank site weight and keff
+    double score = simulation::keff * bank.wgt;
+
+    // Add derivative information for differential tallies.  Note that the
+    // i_nuclide and atom_density arguments do not matter since this is an
+    // analog estimator.
+    if (tally.deriv_ != C_NONE)
+      apply_derivative_to_score(p, i_tally, 0, 0., SCORE_NU_FISSION, score);
+
+    if (!settings::run_CE && eo_filt.matches_transport_groups()) {
+
+      // determine outgoing energy group from fission bank
+      auto g_out = static_cast<int>(bank.E);
+
+      // modify the value so that g_out = 1 corresponds to the highest energy
+      // bin
+      g_out = eo_filt.n_bins() - g_out;
+
+      // change outgoing energy bin
+      simulation::filter_matches[i_eout_filt].bins_[i_bin] = g_out;
+
+    } else {
+
+      double E_out;
+      if (settings::run_CE) {
+        E_out = bank.E;
+      } else {
+        E_out = data::energy_bin_avg[static_cast<int>(bank.E)];
+      }
+
+      // Set EnergyoutFilter bin index
+      if (E_out < eo_filt.bins().front() || E_out > eo_filt.bins().back()) {
+        continue;
+      } else {
+        auto i_match = lower_bound_index(eo_filt.bins().begin(),
+          eo_filt.bins().end(), E_out);
+        simulation::filter_matches[i_eout_filt].bins_[i_bin] = i_match;
+      }
+
+    }
+
+    // Case for tallying prompt neutrons
+    if (score_bin == SCORE_NU_FISSION
+      || (score_bin == SCORE_PROMPT_NU_FISSION && g == 0)) {
+
+      // Find the filter scoring index for this filter combination
+      //TODO: should this include a weight?
+      int filter_index = 0;
+      for (auto j = 0; j < tally.filters().size(); ++j) {
+        auto i_filt = tally.filters(j);
+        auto& match {simulation::filter_matches[i_filt]};
+        auto i_bin = match.i_bin_;
+        filter_index += match.bins_[i_bin] * tally.strides(j);
+      }
+
+      // Update tally results
+      #pragma omp atomic
+      tally.results_(filter_index, i_score, RESULT_VALUE) += score;
+
+    } else if (score_bin == SCORE_DELAYED_NU_FISSION && g != 0) {
+
+      // If the delayed group filter is present, tally to corresponding delayed
+      // group bin if it exists
+      if (tally.delayedgroup_filter_ >= 0) {
+
+        // Get the index of the delayed group filter
+        auto i_dg_filt = tally.filters()[tally.delayedgroup_filter_];
+
+        const DelayedGroupFilter& dg_filt {*dynamic_cast<DelayedGroupFilter*>(
+          model::tally_filters[i_dg_filt].get())};
+
+        // Loop over delayed group bins until the corresponding bin is found
```

Comment:
> Loop over delayed group bins until the corresponding bin is found

Potential Bug: Note that the for loop below will continue to loop even after the corresponding bin is found.  If this is not the intent, I would recommend `break`'ing out of the loop at the end of the `if` block on line 421.